terraform/deployments/cloudfront/main.tf

```diff
@@ -343,3 +348,53 @@ resource "aws_cloudfront_distribution" "assets_distribution" {
     minimum_protocol_version = "TLSv1.2_2021"
   }
 }
+
+import {
+  to = aws_lambda_function.url_rewrite
+  id = "url_rewrite"
+}
+
+resource "aws_iam_role" "basic_lambda_role" {
+  name = "basic_lambda_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": [
+          "lambda.amazonaws.com",
+          "edgelambda.amazonaws.com"
+        ]
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy_attachment" "basic_lambda_attach" {
+  name       = "basic-lambda-attachment"
+  roles      = [aws_iam_role.basic_lambda_role.name]
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+data "archive_file" "url_rewrite" {
+  type        = "zip"
+  source_file = "${path.module}/index.js"
+  output_path = "${path.module}/CloudfrontUrlRewrite.zip"
+}
+
+resource "aws_lambda_function" "url_rewrite" {
```

Comment:
Where is this function actually used? I don't see a reference to it in the cloudfront terraform

Reply:
I've added a `lambda_function_association` that was missing. Does that look like it might work now? @samsimpson1